### PR TITLE
feat(ui): expanded thumbnail and preview features

### LINKED ISSUE #31: Audio Browsing Quality Of Life
In order for this program to be powerful for browsing audio samples in particular.  One should be able to see the waveform preview and  click on a part of the wave form to hear a preview. 

(The video I have attached is of FL Studio's built in file explorer, but this feature is standard in Ableton and many other DAW's and video editing software.)

Every time I Shift+Click on the waveform, it plays the file from where i just clicked.  

I like to search through sample library's, and many of them are completely unorganized and have 8+ samples stored in each audio file.  So I don't really know what's in the file until I've opened it up and listened to the whole thing, which is a headache on windows file explorer, because you need to open up VLC or media player to listen to it. 

Recently I have been experimenting with using alternative DAW's for an alternative workflow experience, but using file explorer for browsing samples sucks.  So I just end up having FL Studio open anyway, which defeats the purpose of trying to take a break from FL Studio.  

The ability to tag and browse samples and preview specific sections of them would be incredibly powerful for any musician and video editor.  

https://github.com/CyanVoxel/TagStudio/assets/62913925/f2211e47-4625-47ae-99d8-e9a064a49384

### LINKED ISSUE #338: [Feature Request]: Make gifs animate in app
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

its  as simple  as  the title make gifs work  in app

(also Apng too but it might work allready and i wouldn't know cause i don't have any apps other than a decomp* thing for Apngs)

*(splits off the other images hidden in the file)

### Solution

make gifs (and maybe Apngs if they  didn't work already)  work in tag studio

(also maybe get  it  to work for those cached thumbnails upcoming feature probably things, but its not as  required as  just making it work  in  the tiny viewer thing)

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by CyanVoxel**
***NOTE:** This PR ended up becoming larger in scope than I originally planned, and normally I would've preferred to separate out some of the specific thumbnail and preview panel additions, however since `thumb_renderer.py` gets a heavy refactor in this it all ended up piling together in one branch. That being said the changes are at least all relevant to each other, even if the scope is a bit bloated.*
***
## This PR introduces several new changes and features for thumbnails and previews:
08242024_thumbnails_screenshot

### Thumbnails
- Add album cover art thumbnails
- Add audio waveform thumbnails for audio files without embedded cover art [#31]
- Add new default file thumbnails, both for generic and specific file types
- Chang the unlinked file icon to better convey its meaning
- Update edge style for thumbnails
- Restyle font previews
- Restyle Blender previews

### Preview Panel
- Add animated GIF playback in the Preview Panel [#338]

### Other UI
- Add option for multiple thumbnail display sizes
- Entry selection colors now follow the system accent color
- All thumbnail/preview graphics now have light/dark theme variants 
- Misc. fixes for certain thumbnail types

Closes #31, Closes #338

### CODE REVIEW

**Review by eivl — CHANGES_REQUESTED** [review]
Approved when comments are resolved.

File: requirements.txt
Comment (seakrueger) [reply requirements.txt:16]: Can confirm, video thumbnails/previews do not work without ffmpeg installed. (Fails to find ffprobe). Could this be something looked into being bundled into pyinstaller? Ffmpeg will be an incredibly useful tool for the future (could probably look into dropping opencv potentially)

File: tagstudio/src/qt/helpers/gradient.py
Comment (CyanVoxel) [reply tagstudio/src/qt/helpers/gradient.py:22]: Might've gotten caught in a find-and-replace

File: tagstudio/src/qt/helpers/file_tester.py
Comment (CyanVoxel) [reply tagstudio/src/qt/helpers/file_tester.py:28]: That's a great question - If so, it wasn't my intention to catch that with this method

File: tagstudio/src/qt/widgets/thumb_renderer.py
Comment (CyanVoxel) [reply tagstudio/src/qt/widgets/thumb_renderer.py:631]: I left a note for this one in the surrounding code:
```py
# NOTE: While a stroke effect is desired, the text
# method only allows for outer strokes, which looks
# a bit weird when rendering fonts.
```
Essentially it's code I would like to use to style the font previews in a way that matches other previews like the audio waveform, but the way the `stroke_width` ends up getting applied on the inside makes the previews harder to read. I left it as a quick way to reimplement it in the case a solution to the stroke alignment was found.

File: tagstudio/src/core/palette.py
Comment (CyanVoxel) [reply tagstudio/src/core/palette.py:16]: `defaultdict` seems like it's well-suited for this, and would indeed replace the need for the try/except block in `get_tag_color`. However, since #332 is doing [some refactor work](https://github.com/TagStudioDev/TagStudio/blob/f436b5c7a1dd989f65173f4bd560ea3e0fbcfa49/tagstudio/src/core/palette.py) in `palette.py`, I feel it would be best to wait until that one is merged to consider switching to `defaultdict`.

[Custom user-defined tag colors](https://github.com/TagStudioDev/TagStudio/pull/372#issuecomment-2307658099) is also a feature I've been wanting to add in the near future (post-#332) that would see this getting shaken up once more. Still, it's good to be aware of `defaultdict` now!

File: tagstudio/src/qt/widgets/thumb_renderer.py
Comment (CyanVoxel) [reply tagstudio/src/qt/widgets/thumb_renderer.py:922]: When going in to fix this I discovered that that neither `FONT_PIXEL_RATIO` nor `ext_font` are actually being used... Welp, takes care of that issue...

**Comment by CyanVoxel** [comment]
Thank you for the review! 

I've pushed some changes that should address most of the comments. The FFmpeg support will probably need to be further resolved/clarified, but for now I've added an additional notice on the README under the installation instructions.

File: tagstudio/src/qt/resource_manager.py
Comment (CyanVoxel) [reply tagstudio/src/qt/resource_manager.py:59]: `ResourceManager` is intended to be a singleton pattern and never to be subclassed. The intention is to not reload duplicate resources for each new instance, when every instance will only ever pull from the same pool of resources.

**Pull request merged**
